docs/adr/adr-009-public-api.md

```diff
@@ -0,0 +1,227 @@
+	SyncHead(ctx context.Context) (*header.ExtendedHeader, error)
+}
+```
+
+##### Shares
+```go
+type SharesModule interface {
+	// GetShare returns the Share from the given data Root at the given row/col coordinates.
+	GetShare(ctx context.Context, dah *Root, row, col int) (Share, error)
+	// GetSharesByNamespace returns all shares of the given nID from the given data Root.
+	GetSharesByNamespace(ctx context.Context, root *Root, nID namespace.ID) ([]Share, error)
+	// SharesAvailable subjectively validates if Shares committed to the given data Root are 
+	// available on the network. 
+	SharesAvailable(ctx context.Context, root *Root) error
+	// ProbabilityOfAvailability calculates the probability of the data square
+	// being available based on the number of samples collected.
+	ProbabilityOfAvailability() float64
+}
+```
+
+##### P2P
+```go
+type P2PModule interface {
+    Info
+    Peers
+    PeerInfo
+
+    Connect
+    Disconnect
+    Connectedness
+    NATStatus
+
+    BlockAdd
+    BlockRm
+    BlockList
+
+    MutualAdd
+    MutualRm
+    MutualList
+
+    BandwidthStats
+    BandwidthPeer
+    BandwidthProtocol
+
+    ResourceState
+    ResourceLimit
+    ResourceSetLimit
+
+    PubSubPeers
+    PubSubScore
+
+    BitSwapStats
+    BitSwapLedger
+
+    DHTFindPeer
+}
+```
+
+##### Wallet
+```go
+type WalletModule interface {
+    New
+    Del
+    Has
+    List
+    Sign
+    Export
+    Import
+    // TODO: Mnemonics stuff?
+}
+```
+
+### NodeModule
+```go
+type NodeModule interface {
+    Type
+    Version
+
+    LogList
+    LogLevelSet
+
+	Start(ctx context.Context) error
+	Stop(ctx context.Context) error
+
+    // TODO: Auth
+}
+```
+
+##### DAS
+// TODO: Should we rebrand DASer to ShareSync?
+```go
+type DASModule interface {
+	// State reports the current state of the DASer.
+    State() das.State
+}
+```
+
+##### State
+```go
+type StateModule interface {
+    // Balance retrieves the Celestia coin balance for the node's account/signer
+    // and verifies it against the corresponding block's AppHash.
+    Balance(ctx context.Context) (*state.Balance, error)
+    // BalanceForAddress retrieves the Celestia coin balance for the given address and verifies
+    // the returned balance against the corresponding block's AppHash.
+    BalanceForAddress(ctx context.Context, addr state.Address) (*state.Balance, error)
+    // SubmitTx submits the given transaction/message to the Celestia network and blocks until 
+	// the tx is included in a block.
+    SubmitTx(ctx context.Context, tx state.Tx) (*state.TxResponse, error)
+    // SubmitPayForData builds, signs and submits a PayForData transaction.
+    SubmitPayForData(ctx context.Context, nID namespace.ID, data []byte, gasLimit uint64) (*state.TxResponse, error)
+    // Transfer sends the given amount of coins from default wallet of the node to the given account address.
+    Transfer(ctx context.Context, to types.Address, amount types.Int, gasLimit uint64) (*state.TxResponse, error)
```

Comment:
`Transfer` is related to `State` but it's also feels like a `wallet` operation.

Reply:
We probably won't have a wallet module for mainnet.

Reply:
We will have the key utility instead 👍🏻